dashboard/src/modules/auth/register/RegisterForm.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useRef, useState } from "react";
+import _axios from "axios";
+import HCaptcha from "@hcaptcha/react-hcaptcha";
+import { Form, Formik } from "formik";
+import { useRouter } from "next/router";
+import {
+  IoCheckmark,
+  IoCheckmarkCircle,
+  IoLockOpenOutline,
+  IoMailOutline,
+  IoPersonOutline,
+} from "react-icons/io5";
+import { ContentBox } from "../../../components/ContentBox";
+import { Button } from "../../../components/forms/Button";
+import { Input } from "../../../components/forms/Input";
+import { useAxios } from "../../../context/AxiosProvider";
+import { API_BASEURL } from "../../../lib/constants";
+import { useHasToken } from "../useHasToken";
+import createRegisterSchema from "./RegistrationSchema";
+
+interface RegisterFormProps {
+  captchaKey: string;
+}
+
+export const RegisterForm: React.FC<RegisterFormProps> = ({ captchaKey }) => {
+  const [schema] = useState(createRegisterSchema());
+  const hCaptchaRef = useRef<HCaptcha>(null);
+  const { replace } = useRouter();
+  const hasToken = useHasToken();
+  const axios = useAxios();
+  const [formError, setFormError] = useState("");
+  const [registerSuccess, setRegisterSuccess] = useState(false);
+
+  useEffect(() => {
+    if (hasToken) {
+      // AuthContext and WaitForAuth will take care
+      // of verifying the access token validity
+      replace("/");
+    }
+  }, [hasToken, replace]);
+
+  return (
+    <ContentBox
+      centered
+      heading={{
+        title: "Welcome",
+        subtitle: "Your regular registration tagline",
+      }}
+      className="min-w-40"
+      error={formError}
+    >
+      <Formik
+        initialValues={{
+          name: "",
+          email: "",
+          password: "",
+          passwordConfirmation: "",
+        }}
+        validationSchema={schema.RegisterSchema}
+        onSubmit={async (formData) => {
+          // Trigger invisible hCaptcha
+          try {
+            if (!hCaptchaRef.current) {
+              throw new Error();
+            }
+
+            const { response } = await hCaptchaRef.current.execute({
+              async: true,
+            });
+
+            if (!response) {
+              throw new Error();
+            }
+          } catch (error) {
+            setFormError("Verification failed");
+            return;
+          }
+
+          // Register user
+          try {
+            await axios(`${API_BASEURL}/users`, {
+              method: "POST",
+              data: {
+                name: {
+                  firstName: formData.name,
+                  lastName: formData.name, // this is wrong, but api requires both
```

Comment:
How is this wrong?

Reply:
There's a field for first name, but not last name. The API requires both first name and last name. 

It is wrong because firstName and lastName use the same value. I've left it out to you to decide which one to edit.

Reply:
Can you add a last name field to the registration page then?